test/integration/dryrun/dryrun_test.go

```diff
@@ -0,0 +1,620 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dryrun
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/features"
+	genericapiserveroptions "k8s.io/apiserver/pkg/server/options"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	utilfeaturetesting "k8s.io/apiserver/pkg/util/feature/testing"
+	cacheddiscovery "k8s.io/client-go/discovery/cached"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/kubernetes/cmd/kube-apiserver/app"
+	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+	"k8s.io/kubernetes/test/integration/framework"
+
+	// install all APIs
+	_ "k8s.io/kubernetes/pkg/master" // TODO what else is needed
+)
+
+// dryrun data for all persisted objects.
+var dryrunData = map[schema.GroupVersionResource]struct {
+	stub                 string // Valid JSON stub to use during create
+	skipDeleteCollection bool   // If set to true, DeleteCollection isn't supported by this resource.
+	skipUpdate           bool   // If set to true, Update is skipped for the resource.
+}{
+	// k8s.io/kubernetes/pkg/api/v1
+	gvr("", "v1", "configmaps"): {
+		stub: `{"data": {"foo": "bar"}, "metadata": {"name": "cm1"}}`,
+	},
+	gvr("", "v1", "services"): {
+		stub:                 `{"metadata": {"name": "service1"}, "spec": {"externalName": "service1name", "ports": [{"port": 10000, "targetPort": 11000}], "selector": {"test": "data"}}}`,
+		skipDeleteCollection: true,
+	},
+	gvr("", "v1", "podtemplates"): {
+		stub: `{"metadata": {"name": "pt1name"}, "template": {"metadata": {"labels": {"pt": "01"}}, "spec": {"containers": [{"image": "fedora:latest", "name": "container9"}]}}}`,
+	},
+	gvr("", "v1", "pods"): {
+		stub: `{"metadata": {"name": "pod1"}, "spec": {"containers": [{"image": "fedora:latest", "name": "container7", "resources": {"limits": {"cpu": "1M"}, "requests": {"cpu": "1M"}}}]}}`,
```

Comment:
you realize that's 1000000 CPUs, right?

Reply:
Actually that shouldn't parse, I thought. should be 1Mi or 1m.

Reply:
The stubs are copy-pasted from `test/integration/etcd/etcd_storage_path_test.go`.  It doesn't really matter, the goal is just to have a valid object for each type.

Reply:
Ah, now I recall-- 1M would be 1*1000*1000, 1Mi would be 1*1024*1024.